### PR TITLE
Improve grammar in Glossary.

### PR DESCRIPTION
**Pull request by Windsooon**
a asynchronous generator -> an asynchronous generator

### CODE REVIEW

File: Doc/whatsnew/3.7.rst [hunk 1]
```diff
@@ -2378,7 +2378,7 @@ Changes in the Python API
   instances.
   (Contributed by Yury Selivanov in :issue:`32327`.)
 
-* :attr:`asyncio.Server.sockets` now returns a copy of the internal list
+* :attr:`asyncio.Server.sockets` now return a copy of the internal list
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I don't think this is correct.

File: Doc/whatsnew/3.7.rst [hunk 2]
```diff
@@ -1223,7 +1223,7 @@ can be set within the scope of a group.
 (Contributed by Serhiy Storchaka in :issue:`25054`.)
 
 Regular expressions compiled with the :const:`re.LOCALE` flag no longer
-depend on the locale at compile time.  Locale settings are applied only
+depends on the locale at compile time.  Locale settings are applied only
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I don't think this is correct.

File: Doc/whatsnew/3.7.rst [hunk 3]
```diff
@@ -1264,7 +1264,7 @@ The :mod:`socket` module now exposes the :data:`socket.TCP_CONGESTION`
 (Contributed by Omar Sandoval in :issue:`26273` and
 Nathaniel J. Smith in :issue:`29728`.)
 
-Support for :data:`socket.AF_VSOCK` sockets has been added to allow
+Support for :data:`socket.AF_VSOCK` sockets have been added to allow
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I have doubts about this.

File: Doc/whatsnew/3.7.rst [hunk 4]
```diff
@@ -1378,7 +1378,7 @@ This is equivalent to passing :data:`subprocess.PIPE` as *stdout* and
 (Contributed by Bo Bayles in :issue:`32102`.)
 
 The ``subprocess.run`` function and the :class:`subprocess.Popen` constructor
-now accept the *text* keyword argument as an alias
+now accepts the *text* keyword argument as an alias
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I have doubts about this.

File: Doc/whatsnew/3.7.rst [hunk 5]
```diff
@@ -1676,7 +1676,7 @@ Added functions :c:func:`PySlice_Unpack` and :c:func:`PySlice_AdjustIndices`.
 :issue:`16500`.)
 
 The ``PyExc_RecursionErrorInst`` singleton that was part of the public API
-has been removed as its members being never cleared may cause a segfault
+has been removed as its members were never cleared may cause a segfault
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I have doubts about this.

File: Doc/whatsnew/3.7.rst [hunk 6]
```diff
@@ -1855,7 +1855,7 @@ Significant speed improvements to alternate constructors for
 constructors when not constructing subclasses. (Contributed by Paul Ganssle
 in :issue:`32403`)
 
-The speed of comparison of :class:`array.array` instances has been
+The speed of comparison of :class:`array.array` instances have been
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I don't think this is correct.

File: Doc/whatsnew/3.7.rst [hunk 7]
```diff
@@ -1930,7 +1930,7 @@ asyncio
 -------
 
 Support for directly ``await``-ing instances of :class:`asyncio.Lock` and
-other asyncio synchronization primitives has been deprecated.  An
+other asyncio synchronization primitives have been deprecated. An
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: I have doubts about this.

**Review by serhiy-storchaka — COMMENTED** [review]
Maybe some of proposed changes are correct, but most of them look incorrect to me. Although I can be wrong.

**Comment by Windsooon** [comment]
Thank you @serhiy-storchaka, I deleted the one I also not sure.

**Comment by Mariatta** [comment]
I look through the first commit, and I agree with @serhiy-storchaka. The current changeset looks good.

**Review by Mariatta — APPROVED** [review]

**Pull request merged**